### PR TITLE
Implement return values on DeleteItem

### PR DESCRIPTION
**Pull request by dagnir**
## Motivation and Context
Requested feature request.

## Description
This commit introduces DynamoDbTable#deleteItemWithResponse() that allows
customers to specify additional parameters on the request such as
ReturnConsumedCapacity to get additional information the service response.

Similar to https://github.com/aws/aws-sdk-java-v2/pull/2710.

Relates to #1918.
## Testing
- Unit tests
- Integration tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by dagnir** [comment]
SonarCloud notes:

 - The duplication reported is unavoidable.
 - [The bug](https://sonarcloud.io/project/issues?id=aws_aws-sdk-java-v2&issues=AXvweu_w5jDQYzNk1uGQ&open=AXvweu_w5jDQYzNk1uGQ&pullRequest=2719) is a false positive since we *do* call `isPresent()` first.

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/DeleteItemEnhancedResponse.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbAsyncTable;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbTable;
+import software.amazon.awssdk.services.dynamodb.model.ConsumedCapacity;
+import software.amazon.awssdk.services.dynamodb.model.DeleteItemResponse;
+import software.amazon.awssdk.services.dynamodb.model.ItemCollectionMetrics;
+
+/**
+ * Defines the elements returned by DynamoDB from a {@code DeleteItem} operation, such as
+ * {@link DynamoDbTable#deleteItem(DeleteItemEnhancedRequest)} and
+ * {@link DynamoDbAsyncTable#deleteItem(DeleteItemEnhancedRequest)}.
```
Comment (cenedhryn) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/DeleteItemEnhancedResponse.java:29]: Should it not be `deleteItemWithResponse`?

Comment (dagnir) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/DeleteItemEnhancedResponse.java:29]: +1

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbAsyncTable.java [hunk 2]
```diff
@@ -236,6 +237,60 @@
...
 
+    /**
+     * Deletes a single item from the mapped table using a supplied primary {@link Key}.
+     * <p>
+     * The additional configuration parameters that the enhanced client supports are defined
+     * in the {@link DeleteItemEnhancedRequest}.
+     * <p>
+     * This operation calls the low-level DynamoDB API DeleteItem operation. Consult the DeleteItem documentation for
+     * further details and constraints.
+     * <p>
```
[the remark this answers is not in the document]
Comment (dagnir) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbAsyncTable.java:248]: +1

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbAsyncTable.java [hunk 3]
```diff
@@ -236,6 +237,60 @@
...
+     * Example:
+     * <pre>
+     * {@code
+     *
+     * MyItem previouslyPersistedItem = mappedTable.delete(DeleteItemEnhancedRequest.builder().key(key).build());
```
[the remark this answers is not in the document]
Comment (dagnir) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbAsyncTable.java:253]: +1

**Review by cenedhryn — APPROVED** [review]

**Pull request merged**